qcodes/instrument_drivers/Spectrum/M4i.py

```diff
@@ -777,22 +786,41 @@ def start_acquisition(self, mV_range, memsize, posttrigger_size=None, verbose=0)
 
         return {'memsize': memsize, 'numch': numch, 'mV_range': mV_range}
 
-    def _transfer_buffer_numpy(self, memsize, numch):
-        """ Transfer buffer to numpy array """
+    def _transfer_buffer_numpy(self, memsize, numch, bytes_per_sample=2):
+        """ Transfer buffer to numpy array
+
+        Args:
+            memsize (int): number of samples to transfer
+            numch (int): number of channels
+            bytes_per_sample (int): specifies the datatype. 2 for int16, 4 for int32
+        Returns:
+            array: transfered data
+
+        """
         # setup software buffer
-        buffer_size = ct.c_int16 * memsize * numch
+        if bytes_per_sample == 2:
+            buffer_size = ct.c_int16 * memsize * numch
+        elif bytes_per_sample == 4:
+            buffer_size = ct.c_int32 * memsize * numch
+        else:
+            raise ValueError('bytes_per_sample should be 2 or 4')
         data_buffer = (buffer_size)()
         data_pointer = ct.cast(data_buffer, ct.c_void_p)
 
         # data acquisition
         self._def_transfer64bit(
-            pyspcm.SPCM_BUF_DATA, pyspcm.SPCM_DIR_CARDTOPC, 0, data_pointer, 0, 2 * memsize * numch)
+            pyspcm.SPCM_BUF_DATA, pyspcm.SPCM_DIR_CARDTOPC, 0, data_pointer, 0, bytes_per_sample * memsize * numch)
         self.general_command(pyspcm.M2CMD_DATA_STARTDMA |
                              pyspcm.M2CMD_DATA_WAITDMA)
 
         # convert buffer to numpy array
         data = ct.cast(data_pointer, ct.POINTER(buffer_size))
-        output = np.frombuffer(data.contents, dtype=ct.c_int16)
+        if bytes_per_sample == 2:
+            output = np.frombuffer(data.contents, dtype=ct.c_int16)
+        elif bytes_per_sample == 4:
+            output = np.frombuffer(data.contents, dtype=ct.c_int32)
+        else:
+            raise ValueError('bytes_per_sample should be 2 or 4')

```

Comment:
what is this exception repeated for? exactly same exception is going to be raised at the very beginning of the method is bytes_per_sample is not 2 or 4. Basically, this else clause is unreachable.